src/test/java/io/openliberty/tools/intellij/it/UIBotTestUtils.java

```diff
@@ -1761,6 +1761,22 @@ public static void waitForLTWTextAreaMessage(RemoteRobot remoteRobot, String mes
                 () -> readAllText(textArea).equals(message));
     }
 
+    /**
+     * Wait for the indexing message to disappear
+     */
+    public static void waitForIndexingToStop(RemoteRobot remoteRobot, int waitTime) {
+        ProjectFrameFixture projectFrame = remoteRobot.find(ProjectFrameFixture.class, Duration.ofSeconds(10));
+        // The Inline progress panel might contain a TextPanel object with text: 'Indexing JDK 17' or other indexing tasks
+        String xPath = "//div[@class='InlineProgressPanel']";
+        Locator progressPanelLocator = byXpath(xPath);
+        JLabelFixture progressPanelFixture = projectFrame.find(JLabelFixture.class, progressPanelLocator, Duration.ofSeconds(10));
+
+        RepeatUtilsKt.waitFor(Duration.ofSeconds(waitTime),
+                Duration.ofSeconds(1),
+                "Waiting for indexing message to disappear e.g. Indexing Java 17...",
+                "Indexing did not appear in the Liberty tool window",
+                () -> progressPanelFixture.findAllText().isEmpty());
+    }
```

Comment:
Is it possible to hit a scenario where we are checking for indexing before indexing has started? If so, how can we handle that?

Reply:
I think your question highlights just how tricky it is to make these UI tests reliable. The `RemoteRobot` is provided by JetBrains, right? I wonder if they have a guide or best practices on how to use it so that you get deterministic results when you apply it to the IDE.

Reply:
Ed or Andy faced this issue when detecting the "empty liberty tools window" test. They waited 2s before checking for indexing before checking the window. I'll try to do the same.

Reply:
The test for the empty liberty tools window actually waits for indexing to start and then waits for it to stop. In the current case we could do a similar thing, check the latest code. Let me know if it has become too complex.